cli/nodeutil.ts

```diff
@@ -174,6 +174,224 @@ export function needsGitCleanAsync() {
         })
 }
 
+export async function getDefaultBranchAsync(): Promise<string> {
+    const b = await gitInfoAsync(["symbolic-ref", "--short", "refs/remotes/origin/HEAD"], undefined, true);
+    if (!b)
+        throw Util.userError("no git remote branch found");
```

Comment:
iirc userError already throws

Reply:
It throws, but typescript typechecking doesn't know that and gets confused when the enclosing function should return a value. I will try updating userError to return `never`.